coa/pkg/apis/v1alpha2/providers/states/redisstate/redisstate.go

```diff
@@ -29,7 +29,37 @@ var rLog = logger.NewLogger("coa.runtime")
 
 const (
 	entryCountPerList = 100
-	separator         = "*"
+	// separator should be disallowed characters in kubernetes custom resource names except for
+	// "." (some resource type contains it), "*" (stands for any string in regex), "_" (used in some prefix like sch_)
+	separator = "|"
+
+	// SetDefaultQuery is the lua script to set the default value
+	// 1. If the etag is not set, set always succeeds
+	// 2. If the etag is set, set succeeds only if
+	// 		1) the etag matches the current etag
+	// 		2) the etag "0" and the first-write flag is not set
+	setDefaultQuery = `
+	local etag = redis.pcall("HGET", KEYS[1], "version");
```

Comment:
Can you add comment to showcase what is the input of KEYS[1], ARGV[1], ARGV[2]?

Reply:
added